specifyweb/context/app_resource.py

```diff
@@ -58,6 +58,42 @@ def get_app_resource(collection, user, resource_name, additional_default=False):
     logger.info('looking for app resource %r for user %s in %s',
                 resource_name, user and user.name, collection and collection.collectionname)
 
+    # Handling for DataObjFormatters to support fallback to defaults
+    if resource_name == 'DataObjFormatters':
```

Comment:
I wonder if this can be generalized to account for all app resources? It doesn't have to be done in this PR since it would extend the scope quite a bit but it's just something to think about since this function already has a different behavior for TypeSearches